File: OffByOne/src/OffByOne.Ql/Ast/Expressions/Expression.cs
namespace OffByOne.Ql.Ast.Expressions
{
    using System.Collections.Generic;

    using OffByOne.Ql.Visitors.Contracts;

    public abstract class Expression : AstNode, IVisitableExpression
    {
        public abstract TResult Accept<TResult, TContext>(IExpressionVisitor<TResult, TContext> visitor, TContext environment)
            where TContext : IEnvironment;

        public abstract ISet<string> GetDependencies();
    }
}

File: Gamma/src/UvA/Gamma/AST/Expression/Expression.java
package UvA.Gamma.AST.Expression;


import UvA.Gamma.AST.ASTNode;
import UvA.Gamma.AST.Expression.Values.Value;
import UvA.Gamma.Visitors.Visitor;

/**
 * Created by Tjarco, 21-03-17.
 */
public abstract class Expression implements ASTNode {
    public abstract Value value();

    public abstract void accept(Visitor visitor);
}
